src/lyra/abstract_domains/container/fulara/fulara_lattice.py

```diff
@@ -168,10 +168,12 @@ def _less_equal(self, other: 'FularaLattice') -> bool:
     def _meet(self, other: 'FularaLattice') -> 'FularaLattice':
         """Point-wise meet of overlapping segments"""
         new_segments = set()
+        other_segments = copy(other.segments)
         for (k1, v1) in self.segments:
-            for (k2, v2) in other.segments:
+            for (k2, v2) in other_segments:
                 if (k1, v1) == (k2, v2):
                     new_segments.add((k1, v1))
+                    other_segments.remove((k2, v2))     # cannot overlap again
```

Comment:
Let me know if that amount of copying is okay. I think it should be more efficient, since it is only a shallow copy and you save a lot of iterations containing meets in the inner loop.